modules/example/src/smithy4s/example/typeclass/EqInterpreter.scala

```diff
@@ -0,0 +1,212 @@
+package smithy4s.example.typeclass
+
+import cats.implicits.{catsSyntaxEq, toContravariantOps}
+import smithy4s.{Schema, _}
+import cats.kernel.Eq
+import smithy4s.capability.EncoderK
+import smithy4s.schema._
+
+import java.util.UUID
+
+object EqInterpreter extends CachedSchemaCompiler.Impl[Eq] {
```

Comment:
@yisraelU already already has an (cleaner) implementation of that [here](https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/compliance-tests/src/smithy4s/compliancetests/internals/eq/EqSchemaVisitor.scala) that could/should be inlined as a bonafide `smithy4s-cats` where `Show` instances could also live.